### PR TITLE
Make OperatingSystem ICloneable and ISerializable

### PR DESCRIPTION
**Pull request by stephentoub**
cc: @danmosemsft 
Contributes to https://github.com/dotnet/corefx/issues/12669
Contributes to https://github.com/dotnet/corefx/issues/12199

### CODE REVIEW

File: src/System.Runtime.Extensions/tests/System/OperatingSystemTests.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.Serialization;
+using System.Runtime.Serialization.Formatters.Tests;
+using Xunit;
+
+namespace System.Tests
+{
+    public static class OperatingSystemTest
```
Comment (justinvp) [inline src/System.Runtime.Extensions/tests/System/OperatingSystemTests.cs]: Nit: `OperatingSystemTests` (plural) to match the file name.

File: src/System.Runtime.Extensions/src/System/OperatingSystem.cs [hunk 2]
```diff
@@ -35,6 +36,43 @@ internal OperatingSystem(PlatformID platform, Version version, string servicePac
...
         }
 
+        private OperatingSystem(SerializationInfo info, StreamingContext context)
+        {
+            SerializationInfoEnumerator enumerator = info.GetEnumerator();
+            while (enumerator.MoveNext())
+            {
+                switch (enumerator.Name)
+                {
+                    case "_version":
```
Comment (jkotas) [inline src/System.Runtime.Extensions/src/System/OperatingSystem.cs]: Can this be just:
```
_version = (Version)info.GetValue("_version", typeof(Version));
_platform = (PlatformID)info.GetValue("_platform", typeof(PlatformID));
_servicePack = info.GetString("_servicePack");
```

I do not think it make sense to replicate all cruft from the old desktop implementation.

File: src/System.Runtime.Extensions/tests/System/OperatingSystemTests.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Runtime.Extensions/tests/System/OperatingSystemTests.cs]: Fixed

File: src/System.Runtime.Extensions/src/System/OperatingSystem.cs (hunk 2 above)
Comment (stephentoub) [inline src/System.Runtime.Extensions/src/System/OperatingSystem.cs]: Yup. Fixed.

**Comment by danmosemsft** [comment]
@dotnet-bot test innerloop windows_nt release build and test (flaky test: https://github.com/dotnet/corefx/issues/12706)

**Comment by danmosemsft** [comment]
@dotnet-bot test innerloop ubuntu14.04 release build and test (another mysterious hang - https://ci.dot.net/job/dotnet_corefx/job/master/job/ubuntu14.04_release_prtest/1700/consoleFull#10606384068101fac1-935e-4556-95f9-7a0e25ca0b52 - started email thread)

**Pull request merged**